  <main role="main">    <h1>Class CertStoreException</h1>   <ul class="inheritance"> <li><a href="../../lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../../lang/throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../../lang/exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="../generalsecurityexception">java.security.GeneralSecurityException</a></li> <li> <ul class="inheritance"> <li>java.security.cert.CertStoreException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><code><a href="../../io/serializable">Serializable</a></code></dd> </dl>  <pre data-language="java">public class CertStoreException
extends GeneralSecurityException</pre> <div>
<p>An exception indicating one of a variety of problems retrieving certificates and CRLs from a <code>CertStore</code>. </p>
<p> A <code>CertStoreException</code> provides support for wrapping exceptions. The <a href="../../lang/throwable#getCause()"><code>getCause</code></a> method returns the throwable, if any, that caused this exception to be thrown. </p>
<p> <b>Concurrent Access</b> </p>
<p> Unless otherwise specified, the methods defined in this class are not thread-safe. Multiple threads that need to access a single object concurrently should synchronize amongst themselves and provide the necessary locking. Multiple threads each manipulating separate objects need not synchronize.</p>
</div> <dl> <dt>Since:</dt> <dd>1.4</dd> <dt>See Also:</dt> <dd>
<a href="certstore"><code>CertStore</code></a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/serialized-form.html#java.security.cert.CertStoreException">Serialized Form</a>
</dd> </dl>      <section role="region">  <h2 id="constructor.summary">Constructors</h2> <table>  <tr> <th scope="col">Constructor</th> <th scope="col">Description</th> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E()">CertStoreException</a>()</code></th> <td> <p>Creates a <code>CertStoreException</code> with <code>null</code> as its detail message.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String)">CertStoreException</a>​(<a href="../../lang/string">String</a> msg)</code></th> <td> <p>Creates a <code>CertStoreException</code> with the given detail message.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String,java.lang.Throwable)">CertStoreException</a>​(<a href="../../lang/string">String</a> msg,
                  <a href="../../lang/throwable">Throwable</a> cause)</code></th> <td> <p>Creates a <code>CertStoreException</code> with the specified detail message and cause.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.Throwable)">CertStoreException</a>​(<a href="../../lang/throwable">Throwable</a> cause)</code></th> <td> <p>Creates a <code>CertStoreException</code> that wraps the specified throwable.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="method.summary">Methods</h2>  <h2 id="methods.inherited.from.class.java.lang.Throwable">Methods declared in class java.lang.<a href="../../lang/throwable">Throwable</a>
</h2> <code><a href="../../lang/throwable#addSuppressed(java.lang.Throwable)">addSuppressed</a>, <a href="../../lang/throwable#fillInStackTrace()">fillInStackTrace</a>, <a href="../../lang/throwable#getCause()">getCause</a>, <a href="../../lang/throwable#getLocalizedMessage()">getLocalizedMessage</a>, <a href="../../lang/throwable#getMessage()">getMessage</a>, <a href="../../lang/throwable#getStackTrace()">getStackTrace</a>, <a href="../../lang/throwable#getSuppressed()">getSuppressed</a>, <a href="../../lang/throwable#initCause(java.lang.Throwable)">initCause</a>, <a href="../../lang/throwable#printStackTrace()">printStackTrace</a>, <a href="../../lang/throwable#printStackTrace(java.io.PrintStream)">printStackTrace</a>, <a href="../../lang/throwable#printStackTrace(java.io.PrintWriter)">printStackTrace</a>, <a href="../../lang/throwable#setStackTrace(java.lang.StackTraceElement%5B%5D)">setStackTrace</a>, <a href="../../lang/throwable#toString()">toString</a></code>  <h2 id="methods.inherited.from.class.java.lang.Object">Methods declared in class java.lang.<a href="../../lang/object">Object</a>
</h2> <code><a href="../../lang/object#clone()">clone</a>, <a href="../../lang/object#equals(java.lang.Object)">equals</a>, <a href="../../lang/object#finalize()">finalize</a>, <a href="../../lang/object#getClass()">getClass</a>, <a href="../../lang/object#hashCode()">hashCode</a>, <a href="../../lang/object#notify()">notify</a>, <a href="../../lang/object#notifyAll()">notifyAll</a>, <a href="../../lang/object#wait()">wait</a>, <a href="../../lang/object#wait(long)">wait</a>, <a href="../../lang/object#wait(long,int)">wait</a></code>  </section>   <div>   <section role="region">  <h2 id="constructor.detail">Constructors</h2>   <h3 id="&lt;init&gt;()">CertStoreException</h3> <pre data-language="java">public CertStoreException()</pre> <p>Creates a <code>CertStoreException</code> with <code>null</code> as its detail message.</p>    <h3 id="&lt;init&gt;(java.lang.String)">CertStoreException</h3> <pre data-language="java">public CertStoreException(String msg)</pre> <p>Creates a <code>CertStoreException</code> with the given detail message. A detail message is a <code>String</code> that describes this particular exception.</p> <dl> <dt>Parameters:</dt> <dd>
<code>msg</code> - the detail message</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.Throwable)">CertStoreException</h3> <pre data-language="java">public CertStoreException(Throwable cause)</pre> <p>Creates a <code>CertStoreException</code> that wraps the specified throwable. This allows any exception to be converted into a <code>CertStoreException</code>, while retaining information about the cause, which may be useful for debugging. The detail message is set to (<code>cause==null ? null : cause.toString()</code>) (which typically contains the class and detail message of cause).</p> <dl> <dt>Parameters:</dt> <dd>
<code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../lang/throwable#getCause()"><code>getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.String,java.lang.Throwable)">CertStoreException</h3> <pre data-language="java">public CertStoreException(String msg,
                          Throwable cause)</pre> <p>Creates a <code>CertStoreException</code> with the specified detail message and cause.</p> <dl> <dt>Parameters:</dt> <dd>
<code>msg</code> - the detail message</dd> <dd>
<code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../lang/throwable#getCause()"><code>getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)</dd> </dl>   </section>  </div>  </main>  <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993, 2020, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/CertStoreException.html" class="_attribution-link">https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/CertStoreException.html</a>
  </p>
</div>
